pkgs/desktops/gnome/extensions/default.nix

```diff
@@ -0,0 +1,136 @@
+{ lib
+, callPackage
+, config
+}:
+let
+  buildShellExtension = callPackage ./buildGnomeExtension.nix { };
+
+  # Index of all scraped extensions (with supported versions)
+  extensionsIndex = lib.importJSON ./extensions.json;
+
+  # A list of UUIDs that have the same pname and we need to rename them
+  # MAINTENANCE:
+  # - Every item from ./collisions.json (for the respective Shell version) should have an entry in here
+  # - Set the value to `null` for filtering (duplicate or unmaintained extensions)
+  # - Sort the entries in order of appearance (check the number in the URL to their extensions.gnome.org page)
+  extensionRenames = {
+    "apps-menu@gnome-shell-extensions.gcampax.github.com" = "applications-menu";
+    "Applications_Menu@rmy.pobox.com" = "frippery-applications-menu";
+
+    "workspace-indicator@gnome-shell-extensions.gcampax.github.com" = "workspace-indicator";
+    "horizontal-workspace-indicator@tty2.io" = "workspace-indicator-2";
+
+    "lockkeys@vaina.lt" = "lock-keys";
+    "lockkeys@fawtytoo" = "lock-keys-2";
+
+    # See https://github.com/pbxqdown/gnome-shell-extension-transparent-window/issues/12#issuecomment-800765381
+    "transparent-window@pbxqdown.github.com" = "transparent-window";
+    "transparentwindows.mdirshad07" = null;
+
+    "floatingDock@sun.wxg@gmail.com" = "floating-dock";
+    "floating-dock@nandoferreira_prof@hotmail.com" = "floating-dock-2";
+
+    # That extension is broken because of https://github.com/NixOS/nixpkgs/issues/118612
+    "flypie@schneegans.github.com" = null;
+  };
+
+  # Take all extensions from the index that match the gnome version, build them and put them into an attrset that is ready to go into nixpkgs
+  produceExtensions = shell-version:
+    lib.trivial.pipe extensionsIndex [
+      # Does a given extension match our current shell version?
+      (builtins.filter
+        (extension: (builtins.hasAttr shell-version extension."shell_version_map"))
+      )
+      # Take in an `extension` object from the JSON and transform it into the correct args to call `buildShellExtension`
+      (map
+        (extension: {
+          inherit (extension) uuid name description link pname;
+          inherit (extension.shell_version_map.${shell-version}) version sha256 metadata;
+        })
+      )
+      # Build them
+      (map buildShellExtension)
+    ];
+
+  # Map the list of extensions to an attrset based on the UUID as key
+  mapUuidNames = extensions:
+    lib.trivial.pipe extensions [
+      (map (extension: lib.nameValuePair extension.extensionUuid extension))
+      builtins.listToAttrs
+    ];
+
+  # Map the list of extensions to an attrset based on the pname as key, which is more human readable than the UUID
+  # We also take care of conflict renaming in here
+  mapReadableNames = extensions: lib.trivial.pipe extensions [
```

Comment:
The argument could be called `extensionList` as well to distinguish it from the attribute set.